src/node/blockstorage.h

```diff
@@ -175,7 +175,7 @@ class BlockManager
      * After this call, it will point to the beginning of the serialized CBlock data, after the separator fields
      * (BLOCK_SERIALIZATION_HEADER_SIZE)
      */
-    bool WriteBlockToDisk(const CBlock& block, FlatFilePos& pos) const;
+    bool WriteBlockToDisk(const CBlock& block, uint32_t block_size, FlatFilePos& pos) const;
```

Comment:
In commit "refactor: Cache block serialized size for consecutive calls" (0cb15046b039d0e6efd2b57075bade46e1a10237)

Would be good to document new `WriteBlockToDisk` `block_size` and `UndoWriteToDisk` `blockundo_size` arguments to clarify how they relate to the `pos` argument. Specifically it would be good to say that size arguments should be set to the size of data without the header, but the `pos` argument will increase by size of the data plus the header.

Reply:
I tried doing that, but noticed that all variants were awkward.
Realized that the code smell is caused by extracting a part of the write methods that isn't a meaningful subset of the functionality: it's tied closely to the only caller (needs the header size twice, recalculated undo serializes size, returns multiple branches, modifies parameter, needs documentation).
So I've recreated the PR completely, inlining these single-use methods, which also solves the serialization size calculations more naturally.